core/src/test/scala/integration/kafka/api/AbstractConsumerTest.scala

```diff
@@ -183,9 +183,13 @@ abstract class AbstractConsumerTest extends BaseRequestTest {
 
   protected def sendRecords(producer: KafkaProducer[Array[Byte], Array[Byte]], numRecords: Int,
                             tp: TopicPartition,
-                            startingTimestamp: Long = System.currentTimeMillis()): Seq[ProducerRecord[Array[Byte], Array[Byte]]] = {
+                            startingTimestamp: Long = System.currentTimeMillis(),
+                            timestampIncrement: Long = -1L): Seq[ProducerRecord[Array[Byte], Array[Byte]]] = {
```

Comment:
Sorry for the delay in the review. I had a day out of the office. It seems to me that the default timestamp increment should be 1. Setting it to negative number (or leaving it as the default of -1) essentially uses 1 as the increment.